akka-projection-dynamodb/src/main/scala/akka/projection/dynamodb/internal/OffsetStoreDao.scala

```diff
@@ -163,18 +187,31 @@ import software.amazon.awssdk.services.dynamodb.model.WriteRequest
         .returnConsumedCapacity(ReturnConsumedCapacity.TOTAL)
         .build()
 
-      val result = client.batchWriteItem(req).asScala
+      val result = writeWholeBatch(req)
 
       if (log.isDebugEnabled()) {
-        result.foreach { response =>
+        result.foreach { responses =>
           log.debug(
             "Wrote latest timestamps for [{}] slices, consumed [{}] WCU",
             offsetsBatch.size,
-            response.consumedCapacity.iterator.asScala.map(_.capacityUnits.doubleValue()).sum)
+            responses.iterator.flatMap(_.consumedCapacity.iterator.asScala.map(_.capacityUnits.doubleValue)).sum)
         }
       }
       result
-        .map(_ => Done)(ExecutionContext.parasitic)
+        .flatMap { responses =>
+          val lastResponse = responses.last
+          if (lastResponse.hasUnprocessedItems) {
+            val unprocessedSliceItems =
+              lastResponse.unprocessedItems.get(settings.timestampOffsetTable).asScala.toVector
```

Comment:
Tests still failing. Need empty checks everywhere. Would be good to refactor to remove the duplication for this too.